File: ruby/enforced.rb
# frozen_string_literal: true

require 'oso-cloud'
require 'sinatra'

api_key = '<please provide your api key here>'
oso = OsoCloud::Oso.new(url: 'https://cloud.osohq.com', api_key:)

get '/organization/:id' do
  org_id = params['id']

  actor = OsoCloud::Value.new(type: 'User', id: 'anonymous')
  resource = OsoCloud::Value.new(type: 'Organization', id: org_id)

  if !oso.authorize(actor, 'view', resource)
    halt 404, "Not Found\n"
  else
    "{ \"id\": \"#{org_id}\" }"
  end
end

set :port, 8000

File: go/enforced.go
package main

import (
	"fmt"
	"net/http"
	"strings"

	oso "github.com/osohq/go-oso-cloud"
)

func main() {
	apiKey := "<please provide your api key here>"
	osoClient := oso.NewClient("https://cloud.osohq.com", apiKey)

	http.HandleFunc("/organization/", func(w http.ResponseWriter, r *http.Request) {
		orgID := strings.TrimPrefix(r.URL.Path, "/organization/")

		actor := oso.Instance{Type: "User", ID: "anonymous"}
		resource := oso.Instance{Type: "Organization", ID: orgID}

		allowed, e := osoClient.Authorize(actor, "view", resource)
		if e != nil || !allowed {
			// Handle authorization failure
			http.Error(w, "Not Found", http.StatusNotFound)
			return
		}

		fmt.Fprintf(w, "{\"id\": \"%s\"}", orgID)
	})

	http.ListenAndServe(":8000", nil)
}
